ArduCopter/ekf_check.cpp

```diff
@@ -113,15 +113,27 @@ void Copter::ekf_check()
 // ekf_over_threshold - returns true if the ekf's variance are over the tolerance
 bool Copter::ekf_over_threshold()
 {
-    // return false immediately if disabled
-    if (g.fs_ekf_thresh <= 0.0f) {
+    // use EKF to get variance
+    float position_var, vel_var, height_var, tas_variance;
+    Vector3f mag_variance;
+    variances_valid = ahrs.get_variances(vel_var, position_var, height_var, mag_variance, tas_variance);
+
+    if (!variances_valid) {
         return false;
     }
 
-    // use EKF to get variance
-    float position_variance, vel_variance, height_variance, tas_variance;
-    Vector3f mag_variance;
-    if (!ahrs.get_variances(vel_variance, position_variance, height_variance, mag_variance, tas_variance)) {
+    uint32_t now_us = AP_HAL::micros();
+    float dt = (now_us - last_ekf_check_us) / 1000000.0f;
+
+    // always update filtered values as this serves the vibration check as well
+    position_var = pos_variance_filt.apply(position_var, dt);
+    vel_var = vel_variance_filt.apply(vel_var, dt);
+    height_var = hgt_variance_filt.apply(height_var, dt);
+
+    last_ekf_check_us = now_us;
```

Comment:
nitpick: could move this up below the calculation of dt just to keep it close to related calculations.